api/src/apps/routes.rs

```diff
@@ -300,19 +350,39 @@ impl<'r> Responder<'r, 'static> for LogsResponse<'r> {
             Some(log_chunk) => log_chunk,
         };
 
-        let from = *log_chunk.until() + chrono::Duration::milliseconds(1);
+        let from = *log_chunk.until() + chrono::Duration::seconds(1);
+
+        let next_logs_url = match self.limit {
+            Some(limit) => format!(
+                "/api/apps/{}/logs/{}?limit={}&since={}",
+                self.app_name,
+                self.service_name,
+                limit,
+                RawStr::new(&from.to_rfc3339()).percent_encode(),
+            ),
+            None => format!(
+                "/api/apps/{}/logs/{}?since={}",
+                self.app_name,
+                self.service_name,
+                RawStr::new(&from.to_rfc3339()).percent_encode(),
+            ),
+        };
 
-        let next_logs_url = format!(
-            "/api/apps/{}/logs/{}/?limit={}&since={}",
-            self.app_name,
-            self.service_name,
-            self.limit,
-            RawStr::new(&from.to_rfc3339()).percent_encode(),
-        );
+        let content_dispositon_value = if self.download {
+            format!(
+                "attachment; filename=\"{}_{}_{}.txt\"",
+                self.app_name,
+                self.service_name,
+                Utc::now().format("%Y%m%d_%H%M%S")
```

Comment:
I think there shouldn't be use of `Utc::now()` and it should be derived from the log data.

```suggestion
                *log_chunk.until().format("%Y%m%d_%H%M%S")
```